ibis/backends/sql/rewrites.py

```diff
@@ -244,6 +255,22 @@ def merge_select_select(_, **kwargs):
     if _.parent.find_below(blocking, filter=ops.Value):
         return _
 
+    if not _.distinct and _.parent.distinct:
+        # SELECT ... FROM (SELECT DISTINCT ...)
+        if all(isinstance(v, ops.Field) for v in _.selections.values()):
+            # Can fuse a trivial subselection after a distinct, like
+            #   SELECT a, b FROM (SELECT DISTINCT ...) -> SELECT DISTINCT a, b, ...
+            distinct = True
+        else:
+            # Other cases aren't safe to fuse as they might introduce non-distinct
+            # columns where applying a distinct would change the result.
+            return _
+    else:
+        # SELECT DISTINCT ... FROM (SELECT DISTINCT ...) -> SELECT DISTINCT ...
+        # SELECT DISTINCT ... FROM (SELECT ...) -> SELECT DISTINCT ...
+        # SELECT ... FROM (SELECT ...) -> SELECT ...
```

Comment:
Might be just me, but I think here you're saying both `distinct`s are `False` and so the merging behavior is unchanged. Is that right?

Reply:
Yes. Just enumerating the 4 possible combinations of outer-query/inner-query `distinct` value.

Reply:
Yeah, it looks like the notation is saying SELECT of SELECT becomes a single SELECT.

How about this?

```suggestion
        # SELECT ... FROM (SELECT ...) -> unchanged
```

Reply:
I just redid the comments completely, hopefully that's clearer.